client/src/Components/SearchForm/index.jsx

```diff
@@ -41,41 +40,51 @@ class SearchForm extends Component {
         this.state.genre === ''
       ) {
         alert('You have not selected any search options');
-      } else {
+      } else if(data) {
         //data.length = 5; - gdy chcę ograniczyć ilość
         this.setState({ books: data });
         this.props.findData(data);
         this.props.handleOpen();
       }
     } catch (error) {
-      alert('The value is not allowed');
-    }
+      console.log(error.response.data.message);
+      this.props.history.push('/login');
+    } 
```

Comment:
W sumie to wam tego nie mówiłem ale jak mamy komponent który odputuje api, to oprócz wyrzucenia w konsoli błędu przydałoby się ustawić state o nazwie np.: error na true a w render zawrzeć coś takiego:

`if (this.state.error) return <Error />`